custom/enikshay/tasks.py

```diff
@@ -0,0 +1,180 @@
+import datetime
+from collections import defaultdict
+import pytz
+from xml.etree import ElementTree
+
+from celery.task import periodic_task
+from celery.schedules import crontab
+from django.conf import settings
+from django.utils.dateparse import parse_datetime
+
+from casexml.apps.case.mock import CaseBlock
+from corehq import toggles
+from corehq.apps.hqcase.utils import submit_case_blocks
+from corehq.apps.fixtures.models import FixtureDataItem
+from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
+from corehq.util.soft_assert import soft_assert
+from dimagi.utils.decorators.memoized import memoized
+
+from .case_utils import get_adherence_cases_between_dates_from_episode, CASE_TYPE_ADHERENCE, CASE_TYPE_EPISODE
+
+
+DOSE_TAKEN_INDICATORS = [
+    'directly_observed_dose',
+    'unobserved_dose',
+    'self_administered_dose',
+]
+DAILY_SCHEDULE_FIXTURE_NAME = 'adherence_schedules'
+DAILY_SCHEDULE_ID = 'schedule_daily'
+SCHEDULE_ID_FIXTURE = 'id'
+
+
+@periodic_task(
+    run_every=crontab(day_of_week=[1], hour=0, minute=0),  # every Monday
+    queue=getattr(settings, 'CELERY_PERIODIC_QUEUE', 'celery')
+)
+def enikshay_adherence_task():
+    domains = toggles.UATBC_ADHERENCE_TASK.get_enabled_domains()
+    for domain in domains:
+        updater = PeriodicCaseUpdater(domain)
+        updater.run()
+
+
+class PeriodicCaseUpdater(object):
+
+    def __init__(self, domain):
+        self.domain = domain
+        self.purge_date = pytz.UTC.localize(datetime.datetime.today() - datetime.timedelta(days=60))
```

Comment:
the intention of this is to define the window in which we don't need to worry about running this task based on your schedule start date right? i'm trying to wrap my head around whether this is ok. is there a specific part of the spec you can refer me to? it feels wrong to me that we don't do anything to reconcile your first 60 days of data.